src/components/ValueSelectorCard/ValueSelectorCard.js

```diff
@@ -0,0 +1,343 @@
+      selectedAssetSymbol,
+    } = this.getMaxBalanceOfSelectedAsset(false, symbol);
+    if (!selectedAssetBalance || !symbol) return null;
+
+    return (
+      <BaseText regular secondary>{selectedAssetBalance} {selectedAssetSymbol} ({amountValueInFiat})</BaseText>
+    );
+  };
+
+  handleFromChange = (value: FormValue) => {
+    const { getFormValue, baseFiatCurrency, rates } = this.props;
+    const { errorMessage, formOptions } = this.state;
+    const formValue = this.form.getValue();
+    const validation = this.form.validate();
+    const currentErrorMessage = get(validation, 'errors[0].message', '');
+
+    const stateUpdates = {};
+    stateUpdates.value = value;
+    if (errorMessage !== currentErrorMessage) {
+      stateUpdates.errorMessage = currentErrorMessage;
+    }
+
+    const { formSelector } = value;
+    const { selector, input: amount } = formSelector;
+    let valueInFiat;
+
+    if (amount && parseFloat(amount) > 0 && !isEmpty(selector)) {
+      const { symbol } = selector;
+      const fiatCurrency = baseFiatCurrency || defaultFiatCurrency;
+      const totalInFiat = parseFloat(amount) * getRate(rates, symbol, fiatCurrency);
+      valueInFiat = symbol ? formatFiat(totalInFiat, baseFiatCurrency) : null;
+    }
+
+    const newOptions = t.update(formOptions, {
+      fields: {
+        formSelector: {
+          config: {
+            inputAddonText: { $set: valueInFiat },
+            customLabel: { $set: this.renderCustomLabel(formSelector?.selector?.symbol) },
+          },
+        },
+      },
+    });
+
+    stateUpdates.formOptions = newOptions;
+
+    this.setState({ ...stateUpdates });
+    getFormValue(formValue?.formSelector);
+  };
+
+  getMaxBalanceOfSelectedAsset = (forSending: boolean, symbol?: string) => {
+    const {
+      balances,
+      baseFiatCurrency,
+      rates,
+      txFeeInfo,
+    } = this.props;
+    const { value } = this.state;
+    const selectedAssetSymbol = symbol || get(value, 'formSelector.selector.symbol');
+    if (!selectedAssetSymbol) return {};
+
+    const rawSelectedAssetBalance = getBalance(balances, selectedAssetSymbol);
+    const selectedAssetBalance = forSending
+      ? calculateMaxAmount(selectedAssetSymbol, rawSelectedAssetBalance, txFeeInfo?.fee, txFeeInfo?.gasToken)
+      : formatAmount(rawSelectedAssetBalance);
+    const fiatCurrency = baseFiatCurrency || defaultFiatCurrency;
+
+    const totalInFiat = parseFloat(rawSelectedAssetBalance) * getRate(rates, selectedAssetSymbol, fiatCurrency);
+    const amountValueInFiat = formatFiat(totalInFiat, baseFiatCurrency);
+
+    return { selectedAssetBalance, amountValueInFiat, selectedAssetSymbol };
+  };
+
+
+  handleUseMax = () => {
+    const { getFormValue } = this.props;
+    const { value, formOptions } = this.state;
+    const { selectedAssetBalance, amountValueInFiat } = this.getMaxBalanceOfSelectedAsset(true);
+    if (!selectedAssetBalance) return;
+    const newValue = { ...value };
```

Comment:
Interesting fact: this will just create a shallow copy of the object, so changing the `formSelector.input` value below will change this in the `value` object as well.
Not saying this is bad in that case, just a side effect :)